src/models/orderbook.rs

```diff
@@ -0,0 +1,128 @@
+use crate::models::Order;
+use anyhow::Result;
+use ethcontract::web3::types::Address;
+use parking_lot::RwLock;
```

Comment:
This is actually a very performant RwLock implementation! Note that there is also one in `std::sync::RwLock`, but the `parking_lot` one is much better from what I've heard.